mu-plugins/rest-api/endpoints/class-wporg-plugins-locale-banner-controller.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+namespace WordPressdotorg\MU_Plugins\REST_API;
+
+use WordPressdotorg\Plugin_Directory\Plugin_Directory;
+use function WordPressdotorg\MU_Plugins\Helpers\Locale\{ get_all_locales_with_subdomain, get_all_valid_locales, get_locale_from_header, get_translated_locales };
+
+/**
+ * Plugins_Locale_Banner_Controller
+ */
+class Plugins_Locale_Banner_Controller extends Base_Locale_Banner_Controller {
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->namespace = 'wporg-plugins/v1';
+		$this->rest_base = 'locale-banner';
+	}
+
+	/**
+	 * Validate the plugin slug.
+	 */
+	public function check_slug( $param ) {
+		return is_string( $param ) && $param && Plugin_Directory::get_plugin_post( $param );
```

Comment:
Is the second`$param` check necessary?

Reply:
It's worth including; it's effectively saying "Is this a string? And it's got text in it? And we're able to find a plugin post for it?" rather than expecting that `get_plugin_post()` properly handles `$slug = ''`.

That being said; I don't think it's needed at all, as the `slug` param should only be coming from the URL rewrite, which should always be a non-zero-length string?

Reply:
That's fair, I copied this one from [the plugin directory code](https://meta.trac.wordpress.org/browser/sites/trunk/wordpress.org/public_html/wp-content/plugins/plugin-directory/api/class-base.php#L62), but it's true that it won't be empty due to the rewrite regex. I've removed it to simplify the check.